client/src/featureform/register.py

```diff
@@ -2479,6 +2479,52 @@ def df_transformation(self,
         self.__resources.append(decorator)
         return decorator
 
+    def ondemand_feature(self,
+                          tags: List[str] = None,
+                          properties: dict = {},
+                          variant: str = "default",
+                          name: str = "",
+                          owner: Union[str, UserRegistrar] = "",
+                          description: str = "",
+                          ):
+        """On Demand Feature decorator.
+
+        Args:
+            variant (str): Name of variant
+            name (str): Name of source
+            owner (Union[str, UserRegistrar]): Owner
+            description (str): Description of on demand feature
+            tags (List[str]): Optional grouping mechanism for resources
+            properties (dict): Optional grouping mechanism for resources
+
+        Returns:
+            decorator (OnDemandFeatureDecorator): decorator
+
+        **Examples**
+        ```python
+        @ff.ondemand_feature()
+        def avg_user_transactions():
+            pass
+        ```
+        """
+
+        if not isinstance(owner, str):
+            owner = owner.name()
+        if owner == "":
+            owner = self.must_get_default_owner()
+    
+        decorator = OnDemandFeatureDecorator(
+            name=name,
+            variant=variant,
+            owner=owner,
+            description=description,
+            tags=tags,
+            properties=properties,
+            status="READY",
```

Comment:
Why not hardcode this in the class inside?

Reply:
wanted to provide the flexibility instead of hardcoding it but I can hardcode it